perun.go

```diff
@@ -228,6 +228,98 @@ type NodeConfig struct {
 
 }
 
+// APIErrorV2 represents the newer version of error returned by node, session
+// and channel APIs.
+//
+// Along with the error message, this error type assignes to each error an
+// error category (that describes how the error should be handled by the
+// client), error code (that identifies specific types of error), additional
+// info (that contains additional data related to the error as key value
+// pairs).
+//
+// TODO: (mano) Once all usage of APIError is replaced with APIErrorV2, the
+// older type needs to be removed and this one to be renamed as APIError.
+// Also, the usage of the string "V2" in the variables and types related to the
+// new error type should be removed.
+type APIErrorV2 interface {
+	Category() ErrorCategory
+	Code() ErrorCode
+	Message() string
+	AddInfo() interface{}
+	Error() string
+}
+
+// ErrorCategory represents the category of the error, which describes how the
+// error should be handled by the client.
+type ErrorCategory int
+
+const (
+	// ParticipantError is caused by one of the channel paritipants not acting
+	// as per the perun protocol.
+	//
+	// To resolve this, the client should negotiate with the peer outside of
+	// this system to act in accordance with the perun protocol.
+	ParticipantError ErrorCategory = iota
+
+	// ClientError is caused by the errors in the request from the client. it
+	// could be errors in arguments or errors in configuration provided by the
+	// client to access the external systems or errors in the state of external
+	// systems not managed by the node.
+	//
+	// To resolve this, the client should provide valid arguments, provide
+	// correct configuration to access the external systems or fix the external
+	// systems; and retry.
+	ClientError
+
+	// ProtocolFatalError is caused when the protocol aborts due to unexpected
+	// failure in external system during execution. It could also result in loss
+	// of funds.
+	//
+	// To resolve this, user should maually inspect the error message and
+	// handle it.
+	ProtocolFatalError
+	// InternalError is caused due to unintended behavior in the node software.
+	//
+	// To resolve this, user should maually inspect the error message and
+	// handle it.
+	InternalError
+)
+
+// String implements the stringer interface for ErrorCategory.
+func (c ErrorCategory) String() string {
+	return [...]string{
+		"Client",
+		"Participant",
+		"Protocol Fatal",
+		"Internal",
+	}[c]
+}
+
+// ErrorCode is a numeric code assigned to identify the specific type of error.
+// The keys in the additional field is fixed for each error code.
+type ErrorCode int
+
+// Error code definitions.
+const (
+	ErrV2PeerResponseTimedout      ErrorCode = 101
```

Comment:
```suggestion
	ErrV2PeerResponseTimedOut      ErrorCode = 101
```
Keep the format same in all places.

Reply:
Updated in 47c015884abb63d42121a1c227363cd3db16f387. Used the suggested format `TimedOut` in all places.